tensorflow_graphics/datasets/pix3d/pix3d.py

```diff
@@ -0,0 +1,251 @@
+"""pix3d dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import os
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+from tensorflow_datasets import features as tfds_features
+
+from tensorflow_graphics.datasets import features as tfg_features
+
+_CITATION = """
+@inproceedings{pix3d,
+  title={Pix3D: Dataset and Methods for Single-Image 3D Shape Modeling},
+  author={Sun, Xingyuan and Wu, Jiajun and Zhang, Xiuming and Zhang, Zhoutong
+  and Zhang, Chengkai and Xue, Tianfan and Tenenbaum, Joshua B and
+  Freeman, William T},
+  booktitle={IEEE Conference on Computer Vision and Pattern Recognition (CVPR)},
+  year={2018}
+}
+"""
+
+_DESCRIPTION = """
+Pix3D is a large-scale dataset of diverse image-shape pairs
+with pixel-level 2D-3D alignment. It has wide applications in shape-related
+ tasks including reconstruction, retrieval, viewpoint estimation, etc.
+
+Pix3D contains 10,069 2D-3D pairs of 395 distinct 3D shapes, categorised into
+nine object categories. Each sample comprises of an image, 3D shape represented
+as (non-watertight) triangle mesh and voxel grid, bounding-box,
+ segmentation mask, intrinsic and extrinsic camera parameters and 2D and 3D key
+ points.
+
+Note:
+  The object and camera poses are provided with respect to the scene, whereas the
+  camera is placed at the origin. Pix3D also provides the features
+  `camera/position_with_respect_to_object` and `camera/inplane_rotation`.
+  Those values are defined in object coordinates and will reproduce an image
+  that is equivalent to the original image under a homography transformation.
+  They are defined for viewer-centered algorithms whose predictions need to be
+  rotated back to the canonical view for evaluations against ground truth shapes.
+  This is necessary as most algorithms assume that the camera is looking at the
+  object's center, the raw input images are usually cropped or transformed
+  before sending into their pipeline.
+"""
+
+_CHECKSUMS_DIR = os.path.normpath(
+    os.path.join(os.path.dirname(__file__), 'checksums/')
+)
+tfds.download.add_checksums_dir(_CHECKSUMS_DIR)
+
+
+class Pix3d(tfds.core.GeneratorBasedBuilder):
+  """Pix3D is a large-scale dataset of diverse image-shape pairs
+  with pixel-level 2D-3D alignment."""
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  TRAIN_SPLIT_IDX = os.path.join(os.path.dirname(__file__),
+                                 'splits/pix3d_train.npy')
+  TEST_SPLIT_IDX = os.path.join(os.path.dirname(__file__),
+                                'splits/pix3d_test.npy')
+
+  CLASS_INDEX = ['bed', 'bookcase', 'chair', 'desk', 'misc', 'sofa', 'table',
+                 'tool', 'wardrobe']
```

Comment:
I think it will be better to add unknown (background) as the first class. `CLASS_INDEX = ['background', 'bed', 'bookcase', ...]`. The reason being that object detection codebases assume 0 as the `backgorund`. And `bed` gets  the class id of 1. Mesh-RCNN also does the same (see https://dl.fbaipublicfiles.com/meshrcnn/pix3d/pix3d_s1_train.json).